server/src/main/java/org/elasticsearch/common/util/BigArrays.java

```diff
@@ -58,15 +58,10 @@ public static long overSize(long minTargetSize, int pageSize, int bytesPerElemen
 
         long newSize;
         if (minTargetSize < pageSize) {
-            newSize = ArrayUtil.oversize((int)minTargetSize, bytesPerElement);
+            newSize = Math.min(ArrayUtil.oversize((int) minTargetSize, bytesPerElement), pageSize);
         } else {
-            newSize = minTargetSize + (minTargetSize >>> 3);
-        }
-
-        if (newSize > pageSize) {
```

Comment:
I wonder if `resizeInPlace` took a number of pages instead of a size and we only used `oversize` for when the size is less than a page. We may not need this method at all in that case. I just feel like we're doing something like `(pages + 1) * pageSize / pageSize` and it feels clumsy.

Reply:
It'd be a bigger change though.